app/javascript/components/components/Header.js

```diff
@@ -1,7 +1,26 @@
 import React, { Component } from 'react'
-import { Nav, NavItem } from 'reactstrap'
+import { Nav, 
+  NavItem, 
+  Navbar, 
+  NavbarBrand, 
+  NavbarToggler, 
+  Collapse, 
+  NavLink, 
+  } from 'reactstrap';
 
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      isOpen: false
+    };
+  }
```

Comment:
If all of your functions use the arrow syntax you will never need to bind. You can add an arrow to line 19 and remove line 14. `toggle = () =>`